codalab/worker/worker_run_state.py

```diff
@@ -344,7 +358,13 @@ def mount_dependency(dependency, shared_file_system):
         for dep in run_state.bundle.dependencies:
             full_child_path = os.path.normpath(os.path.join(run_state.bundle_path, dep.child_path))
             to_mount = []
-            dependency_path = self._get_dependency_path(run_state, dep)
+            if self.shared_file_system:
```

Comment:
Let's revert to using

```
dependency_path = self._get_dependency_path(run_state, dep)
```

Reply:
Ah, I updated this so we don't make an unnecessary call to dependency_manager.get(), which will attempt to acquire the lock.

Reply:
I think it's fine to make an unncessary call to dependency_manager.get() for the sake of simplicity of this PR. This is because this method is `mount_dependency`, which is usually only called once (it's not called super often / repeatedly for the same dependency).